src/wp-includes/interactivity-api/class-wp-interactivity-api.php

```diff
@@ -0,0 +1,930 @@
+
+			/*
+			 * Follows the same logic as Preact in the client and only changes the
+			 * content if the value is a string or a number. Otherwise, it removes the
+			 * content.
+			 */
+			if ( is_string( $result ) || is_numeric( $result ) ) {
+				$p->set_content_between_balanced_tags( esc_html( $result ) );
+			} else {
+				$p->set_content_between_balanced_tags( '' );
+			}
+		}
+	}
+
+	/**
+	 * Processes the `data-wp-router-region` directive.
+	 *
+	 * It renders in the footer a set of HTML elements to notify users about
+	 * client-side navigations. More concretely, the elements added are 1) a
+	 * top loading bar to visually inform that a navigation is in progress
+	 * and 2) an `aria-live` region for accessible navigation announcements.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param WP_Interactivity_API_Directives_Processor $p The directives processor instance.
+	 */
+	private function data_wp_router_region_processor( WP_Interactivity_API_Directives_Processor $p ) {
+		if ( ! $p->is_tag_closer() && ! $this->has_processed_router_region ) {
+			$this->has_processed_router_region = true;
+
+			// Initialize the `core/router` store.
+			$this->state(
+				'core/router',
+				array(
+					'navigation' => array(
+						'message'     => '',
+						'hasStarted'  => false,
+						'hasFinished' => false,
+						'texts'       => array(
+							'loading' => __( 'Loading page, please wait.' ),
+							'loaded'  => __( 'Page Loaded.' ),
+						),
+					),
+				)
+			);
+
+			$callback = static function () {
+				echo <<<HTML
+<style id="wp-interactivity-router_animations">
+	.wp-interactivity-router_loading-bar {
+		position: fixed;
+		top: 0;
+		left: 0;
+		margin: 0;
+		padding: 0;
+		width: 100vw;
+		max-width: 100vw !important;
+		height: 4px;
+		background-color: var(--wp--preset--color--primary, #000);
+		opacity: 0
+	}
+	.wp-interactivity-router_loading-bar.start-animation {
+		animation: wp-interactivity-router_loading-bar-start-animation 30s cubic-bezier(0.03, 0.5, 0, 1) forwards
+	}
+	.wp-interactivity-router_loading-bar.finish-animation {
+		animation: wp-interactivity-router_loading-bar-finish-animation 300ms ease-in
+	}
+
+	@keyframes wp-interactivity-router_loading-bar-start-animation {
+		0% { transform: scaleX(0); transform-origin: 0% 0%; opacity: 1 }
+		100% { transform: scaleX(1); transform-origin: 0% 0%; opacity: 1 }
+	}
+	@keyframes wp-interactivity-router_loading-bar-finish-animation {
+		0% { opacity: 1 }
+		50% { opacity: 1 }
+		100% { opacity: 0 }
+	}
+</style>
+<div
+	class="wp-interactivity-router_loading-bar"
+	data-wp-interactive='{"namespace":"core/router"}'
+	data-wp-class--start-animation="state.navigation.hasStarted"
+	data-wp-class--finish-animation="state.navigation.hasFinished"
+></div>
+<div
+	class="screen-reader-text"
+	aria-live="polite"
+	data-wp-interactive='{"namespace":"core/router"}'
+	data-wp-text="state.navigation.message"
+></div>
+HTML;
```

Comment:
I couldn't make this work for some reason. 

This is the code, but it's not committed to this branch yet: https://github.com/WordPress/wordpress-develop/commit/89de2eb766ad88c7d769f1007a4da5e5cc50acaf

I'll keep trying later, but am I doing something wrong?

Reply:
Oh, looks like when the style's `src` is `false`, it has to be registered first:

```php
		// Enqueues as an inline style.
		wp_register_style( 'wp-interactivity-router_animations', false );
		wp_add_inline_style( 'wp-interactivity-router_animations', $this->router_animation_styles() );
		wp_enqueue_style( 'wp-interactivity-router_animations' );
```

Like this:

https://github.com/WordPress/wordpress-develop/blob/a0fedc97f5aab81cbf35f495bc1c0d1b1be245a4/src/wp-includes/formatting.php#L5873-L5888

I also suggest using all hyphens for the handle instead of an underscore.

Reply:
That worked! Thanks, Weston.

Reply:
Fixed in https://github.com/WordPress/wordpress-develop/pull/5953/commits/8a1caa92a9e0183eaa8f3c33833c2391531f37de and https://github.com/WordPress/wordpress-develop/pull/5953/commits/69afa1346687f53e9193b2ae7953abb3e5f3dbbb.